firefox_puppeteer/ui/windows.py

```diff
@@ -98,7 +98,11 @@ def create_window_instance(self, handle, expected_class=None):
             # Retrieve window type to determine the type of chrome window
             if handle != self.marionette.current_chrome_window_handle:
                 self.switch_to(handle)
+
             window_type = self.marionette.get_window_type()
+            if window_type is None:
+                Wait(self.marionette).until(lambda mn: mn.get_window_type() is not None)
```

Comment:
This is a little hacky but works. Alternatively we could try waiting on the window to load here, but that would be a bigger change.